packages/shared/src/hooks/useOnboardingModal.ts

```diff
@@ -0,0 +1,108 @@
+import { useContext, useEffect, useMemo, useState } from 'react';
+import { MainFeedPage } from '../components/utilities';
+import AnalyticsContext from '../contexts/AnalyticsContext';
+import { Alerts } from '../graphql/alerts';
+import { MyFeedMode } from '../graphql/feed';
+import { OnboardingVersion } from '../lib/featureValues';
+import { LoggedUser } from '../lib/user';
+import usePersistentContext from './usePersistentContext';
+
+interface UseOnboardingModal {
+  myFeedMode: MyFeedMode;
+  isOnboardingOpen: boolean;
+  isCreateMyFeedOpen: boolean;
+  onCloseOnboardingModal: () => void;
+  onInitializeOnboarding: () => void;
+}
+
+interface UseOnboardingModalProps {
+  user?: LoggedUser;
+  alerts: Alerts;
+  isFirstVisit: boolean;
+  onboardingVersion: OnboardingVersion;
+  onFeedPageChanged: (page: MainFeedPage) => unknown;
+}
+
+type ModalCommand = Record<OnboardingVersion, (value: boolean) => unknown>;
+
+const FIRST_TIME_SESSION = 'firstTimeSession';
+const LOGGED_USER_ONBOARDING = 'loggedUserOnboarding';
+
+export const useOnboardingModal = ({
```

Comment:
The logic here is quite hard to follow. Eventually, what we want is simple:
Did the user dismiss onboarding or has done it already? Yes, don't show onboarding. Else, show it

Reply:
This was just extracted from the MainFeedLayout to reduce noise on that component. Probably the reason it became complicated is there are now two options for which one to show. This involved data from the boot query, local storage, and requires being in sync.

With that, I'll see what I can do to refactor this one.

Reply:
After updating the variables, I think it is much clearer now that this is all about onboarding. With that, unfortunately, each part of the code is necessary. Each part of the use effect is necessary as there are two variables to depend on.